web/rest/platform/features/my/activeCalls/getActiveCallsFilters.feature

```diff
@@ -11,6 +11,90 @@ Feature: Retrieve active calls filters
       And the JSON should be equal to:
       """
       {
-          "criteria": "trunks:b1:c2:cr2:dp4:*"
+          "criteria": "trunks:-1"
+      }
+      """
+
+  Scenario: Retrieve platform active calls filter json
+    Given I add Authorization header
+     When I add "Accept" header equal to "application/json"
+      And I send a "GET" request to "/my/active_calls/realtime_filter?b=1&c=2&dp=4"
+     Then the response status code should be 200
+      And the response should be in JSON
+      And the header "Content-Type" should be equal to "application/json"
+      And the JSON should be equal to:
+      """
+      {
+          "criteria": "trunks:b1:c2:dp4:*"
+      }
+      """
+
+  Scenario: Retrieve platform active calls filter json only by carrier
+    Given I add Authorization header
+     When I add "Accept" header equal to "application/json"
+      And I send a "GET" request to "/my/active_calls/realtime_filter?cr=2"
+     Then the response status code should be 200
+      And the response should be in JSON
+      And the header "Content-Type" should be equal to "application/json"
+      And the JSON should be equal to:
+      """
+      {
+          "criteria": "trunks:*:*:cr2:*"
+      }
+      """
+
+  Scenario: Retrieve platform active calls filter json by carrier and direction
+    Given I add Authorization header
+     When I add "Accept" header equal to "application/json"
+      And I send a "GET" request to "/my/active_calls/realtime_filter?cr=2&direction=inbound"
+     Then the response status code should be 200
+      And the response should be in JSON
+      And the header "Content-Type" should be equal to "application/json"
+      And the JSON should be equal to:
+      """
+      {
+          "criteria": "trunks:*:*:cr2:*"
+      }
+      """
+
+  Scenario: Retrieve platform active calls filter json by ddiProvider and direction
+    Given I add Authorization header
+     When I add "Accept" header equal to "application/json"
+      And I send a "GET" request to "/my/active_calls/realtime_filter?dp=1&direction=outbound"
+     Then the response status code should be 200
+      And the response should be in JSON
+      And the header "Content-Type" should be equal to "application/json"
+      And the JSON should be equal to:
+      """
+      {
+          "criteria": "trunks:*:*:dp1:*"
+      }
+      """
```

Comment:
I would return error in these cases: 

```
"criteria": "trunks:-1"
```

While filtering for DDI Provider 1 all calls will be inbound. If you add a direction outbound to the filter, no call should be displayed, but with this response the direction filter is being ignored and Inbound DDI Provider calls will be displayed. Same applies to Carrier and outbound calls: if both filters are set, no call should be displayed.

There was a reference to this in my note:

> If the filter has Carrier and Direction inbound or DDI Provider and Direction outbound (which makes no sense), you can give an error, just like filtering by Carrirer and DDI provider at the same time.